internal/keepers/ocr.go

```diff
@@ -94,7 +126,7 @@ func (k *keepers) Report(ctx context.Context, rt types.ReportTimestamp, _ types.
 
 	// pass the filter to the dedupe function
 	// ensure no locked keys come through
-	keys, err := shuffledDedupedKeyList(attributed, key, k.filter.Filter())
+	keys, err := shuffleUniqueObservations(attributed, key, reportKeysLimit, k.filter.Filter())
```

Comment:
can we refactor this please.

1. Separate function to get median and generate new keys
2. split out reportKeysLimit as a separate step

Reply:
Done